_episodes/29.md

```diff
@@ -0,0 +1,155 @@
+</p>
+
+Trino consists of two types of nodes, a _coordinator_ and _worker_ nodes. The 
+coordinator parses, analyzes, plan, and schedules SQL queries input by the user.
+The workers actually carry out the task by reading the data from the source or
+performing various operations within the task(s) they are assigned.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/2-SPI.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+Trino is able to query over multiple data types by exposing a common interface
+called the SPI (Service Provider Interface) that enables the core engine to
+treat the interactions with each data source the same. Each connector must then
+implement exposing metadata, statistics, data location, and establish one or 
+more connections with an underlying data source.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/3-parser-planner.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+Many of these interfaces are used in the coordinator during the analysis and 
+planning phases. The analyzer, for example, uses the metadata SPI to make sure
+the table in the `FROM` clause actually exists in the data source.  
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/4-distributed-query-plan.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+Once a logical query plan is generated, the coordinator then converts this to a
+distributed query plan that maps actions into stages that contain tasks to be
+run on nodes. Stages model the sequence of events and a directed acyclic graph
+(DAG).
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/5-task-management.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+The coordinator then schedules tasks over the worker nodes as efficiently as 
+possible depending on the physical layout and distribution of the data.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/6-splits.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+Data is split and distributed across the nodes to provide internode parallelism.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/7-parallelism-over-drivers.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+Once this data arrives to the node, it is further divided and processed in 
+parallel.
```

Comment:
need to add the last part ...

workers submitted processed data back to coordinator .. and coordinator provides the results of the query to the user.